src/command_modules/azure-cli-interactive/azure/cli/command_modules/interactive/azclishell/az_completer.py

```diff
@@ -190,10 +196,11 @@ def mute_parse_args(self, text):
         return parse_args
 
     # pylint: disable=too-many-branches
-    def gen_dynamic_completions(self, text):
+    def gen_dynamic_completions(self, text_split):
         """ generates the dynamic values, like the names of resource groups """
         try:  # pylint: disable=too-many-nested-blocks
-
+            args = ' '.join(text_split)
+            args += ' {}'.format(self.unfinished_word) if self.unfinished_word else ''
```

Comment:
`args` sounds like a list, can you change the name?

Also, this doesn't affect the outcome but in the new_word scenario when text = 'some_command ', args becomes 'some_command' (without the space).